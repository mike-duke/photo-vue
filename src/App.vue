<template>
  <div id="app">
    <searchForm @add-search-term="fetchPhotos"></searchForm>
  </div>
</template>

<script>
  import searchForm from './components/searchForm.vue';
  import { keys } from '../keys.js';

  export default {
    name: 'app',
    data() {
      return {
        searchTerm: ''
      }
    },
    methods: {
      fetchPhotos(searchTerm) {
        this.searchTerm = searchTerm;
        const url = 'https://api.unsplash.com/search/photos'
        const extras = '&per_page=25&orientation=squarish';
        const authorization = `Client-ID ${keys.appKey}`
        fetch(`${url}?query=${this.searchTerm}${extras}`, {
          method: 'GET',
          headers: {
            "Authorization": authorization
          }
        })
        .then((response) => response.json())
        .then((data) => console.log(data.results))
      }
    },
    components: {
      searchForm
    }
  };
</script>

<style lang="scss">
  #app {
    font-family: 'Avenir', Helvetica, Arial, sans-serif;
    -webkit-font-smoothing: antialiased;
    -moz-osx-font-smoothing: grayscale;
    text-align: center;
    color: #2c3e50;
    margin-top: 60px;
  }
</style>
