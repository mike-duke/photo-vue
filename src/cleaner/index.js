export const cleanPhotos = (photos) => {
  console.log(photos)
  return photos.map((photo) => {
    const { id, description, photo_tags, user, urls } = photo;
    const { instagram_username, location, name, portfolio_url, profile_image } = user;
    return {
      id, 
      description, 
      photo_tags,
      url: urls.small,
      user: {
        id: user.id,
        instagram_username,
        location,
        name, 
        portfolio_url,
        profile_image: profile_image.small
      }
    }
  });
}