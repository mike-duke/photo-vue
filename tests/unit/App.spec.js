import App from '@/App.vue';
import { mount } from '@vue/test-utils';
import { cleanPhotos } from '@/cleaner/';

jest.mock('@/cleaner/')

describe('App', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = mount(App);
    window.fetch = jest.fn().mockImplementation(() => Promise.resolve({
      json: () => Promise.resolve({
        data: {
          results: []
        }
      })
    }));
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('fetchPhotos', () => {
    it.skip('should call fetch with the correct args when the addSearchTerm event fires', async () => {
      wrapper.vm.$emit('addSearchTerm', 'unicorns');
      expect(window.fetch).toHaveBeenCalledWith('unicorns');
    });

    it.skip('should call cleanPhotos with the correct args', async () => {
      await wrapper.vm.$emit('addSearchTerm', 'rainbows');
      expect(cleanPhotos).toHaveBeenCalledWith([]);
    });

    it('should set the value of photos to the result of cleanPhotos', () => {
      
    })
  });
});