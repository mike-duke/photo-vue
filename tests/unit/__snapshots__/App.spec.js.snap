// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot 1`] = `
<div id="app">
  <section class="search-form">
    <h1 class="app-name">PhotoVue</h1>
    <h3>What would you like to see?</h3>
    <form><input type="text" placeholder="Enter search term" autofocus="autofocus" class="search-input"> <button disabled="disabled">Submit</button></form>
  </section>
</div>
`;
